nmostesting/suites/IS1101Test.py

```diff
@@ -303,585 +247,236 @@ def test_01_03(self, test):
-                )
-                if not valid:
-                    return test.FAIL("Unexpected response from the streamcompatibility API: {}".format(response))
-                if response.status_code != 405:
-                    return test.FAIL("The input {} edid base streamcompatibility request has failed: {}"
-                                     .format(input_id, response))
-            return test.PASS()
+    def test_2(self, test):
+        """Inputs with EDID support return the Effective EDID"""
+        if len(self.edid_inputs) == 0:
+            return test.UNCLEAR("Not tested. No inputs with EDID support found.")
 
-        return test.UNCLEAR("No resources found to perform this test")
+        for inputId in self.is11_utils.sampled_list(self.edid_inputs):
+            self.get_effective_edid(test, inputId)
 
-    def test_01_05_04(self, test):
-        """
-        Verify that there is no base EDID (PUT failure)
-        """
-        if len(self.connected_inputs) != 0 and len(self.not_edid_connected_inputs) != 0:
-            for input_id in self.not_edid_connected_inputs:
-                file = open(VALID_EDID, "rb")
-                response = requests.put(
-                    self.compat_url + "inputs/" + input_id + "/edid/base/",
-                    data=file,
-                    headers={"Content-Type": "application/octet-stream"},
-                )
-                file.close()
-                if response.status_code != 405:
-                    return test.FAIL("The input {} edid base change has failed: {}".format(input_id, response))
-            return test.PASS()
-        return test.UNCLEAR("No resources found to perform this test")
+        return test.PASS()
 
-    def test_01_05_05(self, test):
-        """
-        Verify that there is no base EDID (GET failure)
-        """
-        if len(self.connected_inputs) != 0 and len(self.not_edid_connected_inputs) != 0:
+    def test_3(self, test):
+        """Inputs with Base EDID support handles PUTting and DELETing the Base EDID"""
+        if len(self.base_edid_inputs) == 0:
+            return test.UNCLEAR("Not tested. No inputs with Base EDID support found.")
+
+        for inputId in self.is11_utils.sampled_list(self.base_edid_inputs):
+            # Save the default value of the Effective EDID
+            default_edid = self.get_effective_edid(test, inputId)
+
+            # PUT the Base EDID to the Input
+            valid, response = self.do_request("PUT",
+                                              self.compat_url + "inputs/" + inputId + "/edid/base",
+                                              headers={"Content-Type": "application/octet-stream"},
+                                              data=VALID_EDID)
+            if not valid or response.status_code != 204:
+                return test.FAIL("Unexpected response from "
+                                 "the Stream Compatibility Management API: {}".format(response))
 
```

Comment:
Add reasonable delay after put command.
`start_time = time.time()`
`            while time.time() < start_time + CONFIG.API_PROCESSING_TIMEOUT:`
`                time.sleep(0.2)`

Reply:
This code PUTs and then immediately GETs `/edid/base`. It's clear that handling the new EDID by an external baseband source takes some time but the API is the matter of the Node, and the spec doesn't allow the following situation:
1. value is ABC
2. user sets it to CBA
3. user requests it and gets ABC
4. user waits some time
5. user requests it again and gets CBA